book/Zapp-Pipes/12.-Datasource-for-rokuTV.md

```diff
@@ -0,0 +1,85 @@
+# Data Source Provided (DSP) for Roku-TV
+
+**This documentation is in beta**
+
+The Roku-TV system cannot use the current DSP solution, because the javascript code can't be parsed by Roku. This documentation explain how to solve this problem in a few steps.
+
+Applicaster provide a plugin for Roku-tv, this plugin parse the DSP `manifest.json`. This Roku plugin read the endpoint's described inside the `manifest`. Then the Roku App can call an external (rest) service, in order to receive the feed data from the client API.
+
+This is the steps to follow for enable the data source provided inside roku tv :
+
+1. [Prerequisites](#step1)
+2. [Update The manifest to V2](#step2)
+3. [About the ContextKeys](#step3)
+4. [Create a external service](#step4)
+4. [Rest Service Template (Starter kit example)](#step5)
+4. [About The Node Example](#step6)
+
+#### Step 1) Prerequisites {#step1}
+Before the next step, please make sure you are familiar with the DataSource Provider (DSP) concept, documentation [here](Home.md) .
+
+#### Step 2) Update the manifest to V2 {#step2}
+The Roku system parses the plugin manifest and uses the endpoint provided inside the ```zapp_pipes.v2``` section for call the external rest service.
+
+The following JSON object needs to be added into the DSP plugin manifest file (`plugin-manifest.json`).
+
+***Complete Manifest Real Example*** [Here](https://zapp.applicaster.com/admin/plugin_versions/9909/plugin_manifests/9909)
```

Comment:
Please refer to a manifest file located in the template github repo and not on zapp